### DISCUSSION TITLE
Enums vs Union types

### DISCUSSION THREAD

**Discussion by andy-polhill** [Legal-and-General/canopy#25] [General]
We have an ongoing discussion where we think these should be enums so that they can be referenced in code.
```ts
enum FilterVariant {
  SelectOne,
  SelectMultiple
}
```
Which means in the code we can do this instead of referring to the string value...
```ts
if (this.variant === FilterVariant.SelectMultiple) {
```
This comment is more for awareness, although I think this approach might be better, I'd rather we log it in an issue and then fix throughout the codebase, cc @aaronBery who I think has looked into it too)

_Originally posted by @thatguynamedandy in https://github.com/Legal-and-General/canopy/pull/20#discussion_r502309398_

**Comment by aaronBery** [comment]
This has been stale for a while. On the referenced issue I think I was checking to see if @elenagarrone  could dig up the arguments she had for using types over enums :-)

**Comment by elenagarrone** [comment]
I completely forgot about this! I'll try to write it down tomorrow ;)

**Reply by andy-polhill** [reply]
Just in terms of one of the strengths of an enum that came up in discussion with @paul-request yesterday..

consider this code in a story
```ts
    variant: select(
      'variant',
      ['generic', 'info', 'success', 'warning', 'error'],
      'generic',
    ),
```

with an enum you can do the following, which reduces the repetition quite a lot.
```ts
    variant: select(
      'variant',
      Object.keys(AlertVariant),
      AlertVariant.Generic,
    ),
```

**Reply by elenagarrone** [reply]
I see your point @thatguynamedandy. The only small point is that typescript doesn't generate code for union types unlike enums. But it looks like enums are a bit more powerful so probably worth the slightly bigger bundle size.

**Reply by andy-polhill** [reply]
I suspected that might have been the case, the tricky thing is we often only use that code in story files and not in the application code, so it could be detrimental to the product bundle if it doesn't get used.
